roles/ceph-rgw/tasks/multisite/secondary.yml

```diff
@@ -26,10 +26,17 @@
   run_once: true
 
 - name: create the zone
-  command: "{{ docker_exec_cmd }} radosgw-admin zone create --rgw-zonegroup={{ rgw_zonegroup }} --rgw-zone={{ rgw_zone }} --endpoints=http://{{ rgw_multisite_endpoint_addr }}:{{ radosgw_frontend_port }} --access-key={{ system_access_key }} --secret={{ system_secret_key }} --default"
+  command: "{{ docker_exec_cmd }} radosgw-admin zone create --rgw-zonegroup={{ rgw_zonegroup }} --rgw-zone={{ rgw_zone }} --endpoints={{ rgw_multisite_proto }}://{{ rgw_multisite_endpoint_addr }}:{{ radosgw_frontend_port }} --access-key={{ system_access_key }} --secret={{ system_secret_key }} --default"
   delegate_to: "{{ groups[mon_group_name][0] }}"
   run_once: true
   when:
     - "'No such file or directory' in zonecheck.stderr"
   notify:
     - update period
+
+- name: add other endpoints to the zone
+  command: "{{ docker_exec_cmd }} radosgw-admin zone modify --rgw-zone={{ rgw_zone }} --endpoints {{ rgw_multisite_endpoints_list }}" 
```

Comment:
This will only work if rgw_multisite_endpoints is defined (which is not the default config) so you'll need to add a condition on that task.

Reply:
@dsavineau Added. I also added it to master.yml. Do you know if `run_once: true` will rerun this task if the value of `rgw_multisite_endpoints_list` is changed?